fabric-tree-api-v1/src/main/java/net/fabricmc/fabric/api/tree/v1/FabricTreeRegistry.java

```diff
@@ -0,0 +1,125 @@
+	 * Uses {@link TrunkPlacerTypeInvoker} under the hood.
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link TrunkPlacer}
+	 * @param <T> The owner {@link TrunkPlacer}
+	 * @return Created and registered {@link TrunkPlacerType}
+	 */
+	public static <T extends TrunkPlacer> TrunkPlacerType<T> registerTrunkPlacerType(Identifier id, Codec<T> codec) {
+		return TrunkPlacerTypeInvoker.invokeRegister(id.getNamespace() + "__" + id.getPath(), codec);
+	}
+
+	/**
+	 * Creates a new instance of a {@link FoliagePlacerType}, registers and returns it.<br>
+	 * Uses {@link FoliagePlacerTypeInvoker} under the hood.
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link FoliagePlacer}
+	 * @param <T> The owner {@link FoliagePlacer}
+	 * @return Created and registered {@link FoliagePlacerType}
+	 */
+	public static <T extends FoliagePlacer> FoliagePlacerType<T> registerFoliagePlacerType(Identifier id, Codec<T> codec) {
+		return FoliagePlacerTypeInvoker.invokeRegister(id.getNamespace() + "__" + id.getPath(), codec);
+	}
+
+	/**
+	 * Creates a new instance of a {@link TreeDecoratorType}, registers and returns it.<br>
+	 * Uses {@link TreeDecoratorTypeInvoker} under the hood.
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link TreeDecorator}
+	 * @param <T> The owner {@link TreeDecorator}
+	 * @return Created and registered {@link TreeDecoratorType}
+	 */
+	public static <T extends TreeDecorator> TreeDecoratorType<T> registerTreeDecoratorType(Identifier id, Codec<T> codec) {
+		return TreeDecoratorTypeInvoker.invokeRegister(id.getNamespace() + "__" + id.getPath(), codec);
+	}
+
+	/**
+	 * Creates a new instance of a {@link BlockStateProviderType}, registers and returns it.<br>
+	 * Uses {@link BlockStateProviderTypeInvoker} under the hood.
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link BlockStateProvider}
+	 * @param <T> The owner {@link BlockStateProvider}
+	 * @return Created and registered {@link BlockStateProviderType}
+	 */
+	public static <T extends BlockStateProvider> BlockStateProviderType<T> registerBlockStateProviderType(Identifier id, Codec<T> codec) {
+		return BlockStateProviderTypeInvoker.invokeRegister(id.getNamespace() + "__" + id.getPath(), codec);
+	}
+
+	/**
+	 * A convenience method to avoid conflicts when using {@link IntProviderType#register}.
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link IntProvider}
+	 * @param <T> The owner {@link IntProvider}
+	 * @return Created and registered {@link IntProviderType}
+	 */
+	public static <T extends IntProvider> IntProviderType<T> registerIntProviderType(Identifier id, Codec<T> codec) {
+		return IntProviderType.register(id.getNamespace() + "__" + id.getPath(), codec);
```

Comment:
```suggestion
		return IntProviderType.register(id.toString(), codec);
```

Reply:
Fyi such code will be piped to `Registry.register`, which converts the `toString()` result back as an identifier. This is a stupid process, I know.

Reply:
Yea I was a bit confused there